### PR TITLE
app-crypt/tpm2-tss: Bump to 3.2.0, remove old

### PR DESCRIPTION
**Pull request by salahcoronya**
The distribution tarball app-crypt/tpm2-tss-3.2.0 was generated incorrect by upstream, see https://github.com/tpm2-software/tpm2-tss/issues/231, so the libtpms TCTI  is disabled for this version to work around it.

This is the first release that should actually work with openssl-3.0 (3.1.0 compiles, but TPM2_CreatePrimary doesn't work), which will be needed for tpm2-openssl (not in portage). 

Test require at least curl-7.80.0 (currently not marked stable). 

Integration tests are now enabled. This requires app-crypt/swtpm, which is not keyworded on riscv, so i dropped it for now, A keywording bug can file afterwards.

Also clean up an old version tpm2-tss.

### CODE REVIEW

File: app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild [hunk 1]
```diff
@@ -11,30 +11,32 @@ SRC_URI="https://github.com/tpm2-software/${PN}/releases/download/${PV}/${P}.tar
...
 
 RDEPEND="acct-group/tss
 	acct-user/tss
-	fapi? (
-		dev-libs/json-c
-		net-misc/curl
-	)
-	gcrypt? ( dev-libs/libgcrypt:0= )
-	openssl? ( dev-libs/openssl:0= )"
+	fapi? ( dev-libs/json-c
```
Comment (thesamesam) [inline app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild:24]: Not subscribing to json-c subslot? Do we not link against it?

File: app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild [hunk 2]
```diff
@@ -11,30 +11,32 @@ SRC_URI="https://github.com/tpm2-software/${PN}/releases/download/${PV}/${P}.tar
 
 LICENSE="BSD-2"
 SLOT="0"
-KEYWORDS="amd64 arm arm64 ppc64 x86"
-IUSE="doc +fapi gcrypt +openssl static-libs test"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+IUSE="doc +fapi  +openssl mbedtls static-libs test"
```
Comment (thesamesam) [inline app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild:15]: whitespace

File: app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild (hunk 1 above)
Comment (salahcoronya) [reply app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild]: I did not know it had a subslot. Fixed.

File: app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild (hunk 2 above)
Comment (salahcoronya) [reply app-crypt/tpm2-tss/tpm2-tss-3.2.0.ebuild]: fixed

**Pull request closed without merging**